nilearn/decomposition/_base.py

```diff
@@ -0,0 +1,533 @@
+                             % self.__class__.__name__)
+        params = get_params(self.__class__, self)
+        # Remove the mask-computing params: they are not useful and will
+        # just invalid the cache for no good reason
+        for name in ('mask_img', 'mask_args'):
+            params.pop(name, None)
+        data, _ = self._cache(session_pca, func_memory_level=2,
+                              ignore=['verbose', 'memory', 'copy'])(
+                                  imgs, self.mask_img_,
+                                  params,
+                                  memory_level=self.memory_level,
+                                  memory=self.memory,
+                                  verbose=self.verbose,
+                                  confounds=confounds,
+                                  copy=copy,
+                                  sample_mask=sample_mask,
+                                  n_components=self.n_components,
+                                  random_state=self.random_state
+                                  )
+        return data
+
+    def transform_imgs(self, imgs_list, confounds=None, copy=True, n_jobs=1):
+        ''' Prepare multi subject data in parallel
+
+        Parameters
+        ----------
+
+        imgs_list: list of Niimg-like objects
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+            List of imgs file to prepare. One item per subject.
+
+        confounds: list of confounds, optional
+            List of confounds (2D arrays or filenames pointing to CSV
+            files). Must be of same length than imgs_list.
+
+        copy: boolean, optional
+            If True, guarantees that output array has no memory in common with
+            input array.
+
+        n_jobs: integer, optional
+            The number of cpus to use to do the computation. -1 means
+            'all cpus'.
+        '''
+        if not hasattr(self, 'mask_img_'):
+            raise ValueError('It seems that %s has not been fitted. '
+                             'You must call fit() before calling transform().'
+                             % self.__class__.__name__)
+        params = get_params(self.__class__, self)
+
+        target_fov = None
+        if self.target_affine is None:
+            # Force resampling on first image
+            target_fov = 'first'
+
+        niimg_iter = _iter_check_niimg(imgs_list, ensure_ndim=None,
+                                       atleast_4d=False,
+                                       target_fov=target_fov,
+                                       memory=self.memory,
+                                       memory_level=self.memory_level,
+                                       verbose=self.verbose)
+
+        func = self._cache(session_pca, func_memory_level=2,
+                           ignore=['verbose', 'memory', 'copy'])
+        if confounds is None:
+            confounds = itertools.repeat(None, len(imgs_list))
```

Comment:
I ll leav it here as this is the same code as in MultiNiftiMasker. Either we change everything or we keep to this (this is not dangerous as we know that we use it only once).